resources/js/src/app/components/itemList/filter/ItemFilter.js

```diff
@@ -63,5 +69,19 @@ Vue.component("findologic-item-filter", {
                 name: parentCategory.name + '_' + subCategory.name
             };
         },
+
+        /**
+         * @returns {boolean}
+         */
+        shouldShowFilter() {
+            return this.facet.id !== 'cat' ||
+                this.facet.id === 'cat' && typeof this.showCategoryFilter === 'undefined' ||
+                this.facet.id === 'cat' && this.showCategoryFilter;
+        },
+
+        shouldShowCategoryFilter() {
```

Comment:
Great that you also added typehints to most methods. Here you could also add one :)